src/include/misc.h

```diff
@@ -251,21 +251,28 @@
 
 /* Timestamp type and helper macros. */
 #if WT_TIMESTAMP_SIZE > 0
-#define	HAVE_TIMESTAMPS 1
+#define	HAVE_TIMESTAMPS
 #else
-#undef HAVE_TIMESTAMPS
+#undef	HAVE_TIMESTAMPS
 #endif
 
 #ifdef HAVE_TIMESTAMPS
-#define	WT_TIMESTAMP(x) (x)
-typedef uint8_t wt_timestamp_t[WT_TIMESTAMP_SIZE];
-#define	WT_DECL_TIMESTAMP(x) wt_timestamp_t x;
+struct __wt_timestamp_t {
+	union {
+		uint64_t val;
+		uint8_t ts[WT_TIMESTAMP_SIZE];
+	};
+};
```

Comment:
Do we need both the `struct` and the `union`, or would just the `union` suffice?